src/envoy/core.clj

```diff
@@ -112,6 +127,35 @@
        (tools/props->map serializer)
        (get-in (tools/cpath->kpath offset))))
 
+(defn- update-consul
+    [kv-path m & [{:keys [serializer] :or {serializer :edn} :as ops}]]
+    (let [[consul-url sub-path]  (string/split kv-path #"kv" 2)
+          update-kv-path (str consul-url "kv")
+          kpath (tools/cpath->kpath sub-path)
+          stored-map (reduce (fn [acc [k v]]
+                               (merge acc (consul->map
+                                            (str kv-path "/" (name k))
+                                            {:serializer serializer})))
+                               {} m)
+         ;;to update correctly seq we need to pre-serialize map
+          [to-add to-remove _] (diff (tools/serialize-map m serializer)
+                                    (tools/serialize-map (get-in stored-map kpath) serializer))]
+         ;;add
+         (doseq [[k v] (tools/map->props to-add serializer)]
+             (put (str kv-path "/" k) (str v) (dissoc ops :serializer :update)))
+         ;;remove
+         (doseq [[k v] (tools/map->props to-remove serializer)]
+            (when (nil? (get-in to-add (tools/cpath->kpath k) nil))
+                @(http/delete (str kv-path "/" k))))))
+
+(defn map->consul
+  [kv-path m & [{:keys [serializer update] :or {serializer :edn udpate false} :as ops}]]
```

Comment:
booleans in Clojure usually end with a question mark: i.e. `update?`
also "update" does not convey the meaning what what the function does, `overwrite?` could be a better name